storm-compatibility/src/java/backtype/storm/topology/BoltDeclarerImpl.java

```diff
@@ -35,13 +35,18 @@ public BoltDeclarerImpl(com.twitter.heron.api.topology.BoltDeclarer delegate) {
   @Override
   @SuppressWarnings({"rawtypes", "unchecked"})
   public BoltDeclarer addConfigurations(Map conf) {
-    delegate.addConfigurations((Map<String, Object>) conf);
+    // Translate config to heron config and then apply.
```

Comment:
Currently the translation occurs at the end of the step, in StormSubmitter. Any reason  not to do it there? That way there will be just one centralized place of doing it

Reply:
The translation in StormSubmitter processes topology level configs. Components have their own configs and component level configs are merged later when Spouts and Bolts are created.


Without this change, the raw Storm configs for Spouts and Bolts are passed into Heron layer. As the result, TOPOLOGY_TICK_TUPLE_FREQ_MS is missing in Bolt level because it is generated in the translation by Config.setTickTupleFrequency().

Reply:
Doing all the translation in StormSubmitter.submitTopology() is one option, but it would require iterating all components and replacing their configs(so there will be heron configs in storm objects, which is not ideal). Overall, doing translation in *DeclarerImpl seems to be cleaner to us.